src/language/extension-types.md

```diff
@@ -0,0 +1,315 @@
+  var diff2 = num.value - 2; // Ok: Can access representation object with reference.
+  var sum2 = num + 2; // Error: Can't assign 'int' to parameter type 'NumberE'. 
+  
+  List<NumberE> numbers = [
+    NumberE(1), 
+    num.i, // Ok: 'i' getter returns type 'NumberE'.
+    1, // Error: Can't assign 'int' element to list type 'NumberE'.
+    1 as NumberE, // Ok: Dynamic type cast recognizes representation type.
+  ];
+}
+```
+
+#### Transparency
+
+An extension type that [implements](#implements) its representation type
+is considered *transparent* because it gives the extension type access
+to *all* members of the representation type. Otherwise,
+only the members declared in the extension type body are available to it
+(considered a [*somewhat protected*](#type-considerations) extension type).
+
+```dart
+extension type NumberT(int value) 
+  implements int {
+  // Doesn't explicitly declare any members of 'int'.
+  NumberT get i => NumberT(value);
+}
+
+void main () {
+  var num = NumberT(1);
+  
+  // All OK: Representation type interface is available to extension type:
+  int numT = NumberT(2);
+  num.i - num; numT + num; num - 1; 2 + num;
+  
+  // Error: Extension type interface is not available to representation type:
+  numT.i;
+}
+```
+
+Transparency is a conceptual decision for the extension type creator to consider.
+A transparent extension type provides an *extended* interface to an existing type.
+You can invoke all the members of the representation type,
+plus any auxillary members you declare. The new interface is available
+to instances of the extension type.
+
+The inverse to transparency is restricting all members of the representation,
+providing a totally *different* interface to an existing type.
+This means simply not including it in the extension type's implements clause,
+and not declaring any of its members in its definition.
+This is as close as you can get to the complete protection of a wrapper class.
+
+## Type considerations
+
+Extension types are a compile-time wrapping construct.
+So, at runtime, there is absolutely no trace of the extension type.
+Any type query or similar runtime expression works on the representation type.
+
+This makes extension types an *unsafe* abstraction,
+because you can always find out the representation type at runtime
+and access the underlying object.
+
+Dynamic type tests (`e is T`), casts (`e as T`),
+and other runtime type queries (like `switch (e) ...` or `if (e case ...)`)
+all evaluate to the underlying representation type. 
+That's true when the static type of `e` is an extension type,
+and when testing against an extension type (`case MyExtensionType(): ... `).
+
+```dart
+void main() {
+  var num = NumberE(1);
+  
+  if (num case int x) {  // True because runtime type is representation type.
+    print(x);
+  }
+  
+  switch (num) {
+    case (int x): print(x);  // Matches because runtime type is representation type.
```

Comment:
Maybe omit the parens:
```suggestion
    case int x: print(x);  // Matches because runtime type is representation type.
```

It's just going to make some readers speculate about why those parentheses are included, and whether it's even allowed. ;-)